R/qra_fit_convenience.R

```diff
@@ -336,6 +322,46 @@ build_covid_ensemble <- function(
     }
   }
 
+  # Determine locations according to spatial_resolution and hub
+  spatial_resolution <- match.arg(
+    spatial_resolution, 
+    choices = c("county", "state", "national", "state_national", "state_no_territories", "euro_countries"),
+    several.ok = TRUE
+  )
+  hub <- match.arg(hub, choices = c("US", "ECDC"))
+
+  all_locations <- c()
+
+  for (sr in spatial_resolution) {
+    if (hub == "US") {
+      locations <- covidHubUtils::hub_locations
+      if (sr == "county") {
+        locations <- locations %>% dplyr::filter(geo_type == "county") 
+      } else if (sr == "state") {
+        locations <- locations %>% dplyr::filter(geo_type == "state", fips != "US") 
+      } else if (sr == "national") {
+        locations <- locations %>% dplyr::filter(fips == "US") 
+      } else if (sr == "state_national") {
+        locations <- locations %>% dplyr::filter(geo_type == "state") 
+      } else if (sr == "state_no_territories") {
+        locations <- locations %>% dplyr::filter(geo_type == "state", fips <= "56", fips != "US")
+      } else {
+        stop("Undefined spatial resolution for US hub")
+      }
+      locations <- locations %>% dplyr::pull(fips)
+    } else if (hub == "ECDC") {
+      if (sr %in% c("euro_countries", "national")) {
+        locations <- covidHubUtils::hub_locations_ecdc$location
+      } else {
+        stop("Undefined spatial resolution for ECDC hub")
+      }
+    } else {
+      stop("Unknown hub specification")
+    }
+    all_locations <- c(all_locations, locations)
```

Comment:
I wonder if we should insert a `unique` here? For example, if the user calls with `spatial_resolution = c("state", "state_national")` (I don't think we would ever do this... so maybe not important.)
```suggestion
    all_locations <- unique(c(all_locations, locations))
```